desktop-widgets/subsurfacewebservices.cpp

```diff
@@ -387,8 +387,7 @@ SubsurfaceWebServices::SubsurfaceWebServices(QWidget *parent, Qt::WindowFlags f)
 
 	if (userid.isEmpty() &&
 	    !same_string(prefs.cloud_storage_email, "") &&
-	    !same_string(prefs.cloud_storage_password, "") &&
-	    GpsLocation::hasInstance())
+	    !same_string(prefs.cloud_storage_password, ""))
 		userid = GpsLocation::instance()->getUserid(prefs.cloud_storage_email, prefs.cloud_storage_password);
```

Comment:
Now that you no longer check if there's an instance, the code will crash if there is no instance.
I wouldn't call that code "useless"...

Reply:
commit removed.